test/jdk/java/awt/LightweightComponent/LightWeightTabFocus/LightWeightTabFocus.java

```diff
@@ -0,0 +1,154 @@
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 4095214
+ * @summary Test change of focus on lightweights using the tab key
+ * @key headful
+ * @run main LightWeightTabFocus
+ */
+
+import java.awt.Button;
+import java.awt.Component;
+import java.awt.Dimension;
+import java.awt.EventQueue;
+import java.awt.FlowLayout;
+import java.awt.Frame;
+import java.awt.Graphics;
+import java.awt.Point;
+import java.awt.Robot;
+import java.awt.event.FocusEvent;
+import java.awt.event.FocusListener;
+import java.awt.event.InputEvent;
+import java.awt.event.KeyEvent;
+
+public class LightWeightTabFocus {
+    private static Frame f;
+    private static LightweightButton btn1;
+    private static Button btn2;
+    private static Robot robot;
+    private static volatile Point point;
+    private static Point loc;
+
+    public static void main(String[] args) throws Exception {
+        robot = new Robot();
+        robot.setAutoDelay(100);
+        try {
+            EventQueue.invokeAndWait(() -> createUI());
+            robot.delay(1000);
+            EventQueue.invokeAndWait(() -> {
+                loc = f.getLocation();
+                point = btn2.getLocation();
+            });
+            robot.mouseMove(loc.x + point.x, loc.y + point.y);
+            robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+            // First TAB
+            robot.keyPress(KeyEvent.VK_TAB);
+            robot.keyRelease(KeyEvent.VK_TAB);
+            if (!btn1.hasFocus()) {
+                new RuntimeException("First tab failed");
+            }
+            // Second TAB
+            robot.keyPress(KeyEvent.VK_TAB);
+            robot.keyRelease(KeyEvent.VK_TAB);
+            if (!btn2.hasFocus()) {
+                new RuntimeException("Second tab failed");
+            }
+            // First SHIFT+TAB
+            robot.keyPress(KeyEvent.VK_SHIFT);
+            robot.keyPress(KeyEvent.VK_TAB);
+            robot.delay(100);
+            robot.keyRelease(KeyEvent.VK_TAB);
+            robot.keyRelease(KeyEvent.VK_SHIFT);
+            if (!btn1.hasFocus()) {
+                new RuntimeException("First shift+tab failed");
+            }
+            // Second SHIFT+TAB
+            robot.keyPress(KeyEvent.VK_SHIFT);
+            robot.keyPress(KeyEvent.VK_TAB);
+            robot.delay(100);
+            robot.keyRelease(KeyEvent.VK_TAB);
+            robot.keyRelease(KeyEvent.VK_SHIFT);
+            if (!btn2.hasFocus()) {
+                new RuntimeException("Second shift+tab failed");
+            }
+
+        } catch (Exception e) {
+            e.printStackTrace();
+        } finally {
```

Comment:
redundant catch block...guess you mentioned muscle memory and will be removed in open...but seems like it's forgotten..